src/detect-engine.c

```diff
@@ -1955,6 +1955,10 @@ static int DetectEngineInspectRulePayloadMatches(
                 SCLogDebug("SIG_FLAG_REQUIRE_STREAM_ONLY, so no match");
                 return DETECT_ENGINE_INSPECT_SIG_NO_MATCH;
             }
+            if (s->flags & SIG_FLAG_REQUIRE_STREAM_ONLY) {
+                SCLogDebug("SIG_FLAG_REQUIRE_STREAM_ONLY, so no match");
+                return false;
+            }
```

Comment:
What's the difference b/w this and the branch right above this?

Reply:
rebase artifact

Reply:
actually this patch was already applied :-/ 58c7a438ed22e82f7ed1fd853676fd789ff82f05

Reply:
No, just that this change shouldn't be a part of this PR,  I think.